### PR TITLE
Add RPI2 and MBM pin numbers to Lighting via pins_arduino.h

### PR DESCRIPTION
**Pull request by MahmoudGSaleh**
Some cleanup in the package as well, including updating the nuspec
version number and dependency to make upgrades easy and moving DMAP
Errors to a its own cpp file.
Optimizing DMAP lock/unlock async code path.

### CODE REVIEW

File: source/DmapSupport.cpp [hunk 1]
```diff
@@ -383,20 +348,14 @@ HRESULT GetControllerLock(HANDLE & handle)
...
 
-        auto workItem = ref new WorkItemHandler(
-            [&device, &hr, &ioControlCompleted](IAsyncAction^ workItem)
+        create_task(device->SendIOControlAsync(LockCode, nullptr, nullptr)).then([&ioControlCompleted, &hr](UINT result)
         {
-            create_task(device->SendIOControlAsync(LockCode, nullptr, nullptr)).then([&ioControlCompleted, &hr](UINT result)
-            {
-                // TODO: Is this correct?
-                hr = result;
+            // TODO: Is this correct?
```
Comment (ooeygui) [inline source/DmapSupport.cpp]: Any chance we can get rid of this todo?

File: source/DmapSupport.cpp [hunk 2]
```diff
@@ -428,20 +387,14 @@ HRESULT ReleaseControllerLock(HANDLE & handle)
...
 
-        auto workItem = ref new WorkItemHandler(
-            [&device, &hr, &ioControlCompleted](IAsyncAction^ workItem)
+        create_task(device->SendIOControlAsync(UnlockCode, nullptr, nullptr)).then([&ioControlCompleted, &hr](UINT result)
         {
-            create_task(device->SendIOControlAsync(UnlockCode, nullptr, nullptr)).then([&ioControlCompleted, &hr](UINT result)
-            {
-                // TODO: Is this correct?
-                hr = result;
+            // TODO: Is this correct?
```
Comment (ooeygui) [inline source/DmapSupport.cpp]: Can we get rid of this one?

File: source/Microsoft.IoT.Lightning.nuspec [hunk 3]
```diff
@@ -15,7 +15,7 @@
     <copyright>Copyright 2015</copyright>
     <tags>lightning arduino microsoft native nativepackage</tags>
     <dependencies>
-        <dependency id="Microsoft.IoT.SDKFromArduino" version="(,1.0.1]"/>
+        <dependency id="Microsoft.IoT.SDKFromArduino" version="[1.0.1,)"/>
```
Comment (ooeygui) [inline source/Microsoft.IoT.Lightning.nuspec:18]: Oops

File: source/DmapSupport.cpp (hunk 1 above)
Comment (MahmoudGSaleh) [inline source/DmapSupport.cpp]: Yes, we'll use the returned value

File: source/Microsoft.IoT.Lightning.nuspec (hunk 3 above)
Comment (MahmoudGSaleh) [inline source/Microsoft.IoT.Lightning.nuspec:18]: Actually, this is intended, since the original version is already on GitHub.

**Pull request merged**